src/lib.rs

```diff
@@ -86,3 +87,76 @@ pub mod __macros_internal {
         pub last_ptr: u64,
     }
 }
+
+#[derive(Deref, DerefMut)]
+pub struct AppWrapper(send_wrapper::SendWrapper<bevy::app::App>);
+impl AppWrapper {
+    pub fn new() -> AppWrapper {
+        AppWrapper(send_wrapper::SendWrapper::new(bevy::app::App::default()))
+    }
+}
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash, Default)]
+pub struct HotReloadUpdate;
+
+pub trait ReloadableAppExt {
+    /// Call this with plugins and systems and it will auto-add and remove systems in the `Update` schedule to your running app
+    fn reloadable(
+        &mut self,
+        func: impl FnMut(AppWrapper) -> AppWrapper + Send + Sync + 'static,
+    ) -> &mut App;
+}
+
+impl ReloadableAppExt for App {
+    fn reloadable(
+        &mut self,
+        mut func: impl FnMut(AppWrapper) -> AppWrapper + Send + Sync + 'static,
```

Comment:
If we pass `&mut AppWrapper` instead, the API would look a bit nicer, as nothing has to be returned :)

Reply:
Edit: didn't manage to do this on my own because of lifetimes, grrr